src/shared/lib/user-benefits-api.ts

```diff
@@ -0,0 +1,21 @@
+import { z } from 'zod';
```

Comment:
I was playing around in zod, and came across this better way of making the schema!

```ts
import z from 'zod';

// list of benefits we're looking for
const BenefitEnum = z.enum(['adFree', 'allowRejectAll', 'hideSupportMessaging']);

const responseSchema = z.object({
  benefits: z
    .array(z.string()) // Accept any array of strings
    .transform((values) => values.filter((val) => BenefitEnum.safeParse(val).success)) // Filter out unknown values based on the BenefitEnum
});

// Example usage
const response = {
  benefits: ['adFree', 'allowRejectAll', 'unknownBenefit'] // Contains an invalid value
};

const parsedResponse = responseSchema.parse(response); // Will remove 'unknownBenefit'
console.log(parsedResponse); // { benefits: [ 'adFree', 'allowRejectAll' ] }
```

Reply:
Thanks. I've implemented this